### ISSUE TITLE
connection problem

### ISSUE THREAD

**Issue by pavel-demin** [sefffal/redpitaya-multichannelanalyser#5]
Hi Will,

Just found another problem. If I click on 'Will Thompson' and then on the 'Go Back' button in a browser, I get a 'Connection error' message. The same error message appears when I go to any external link from the application and then return. Tested with Firefox and Internet Explorer.

Best regards,

Pavel

**Comment by sefffal** [comment]
Hi Pavel,
Thanks for finding this bug.
I think the websocket is disconnecting when you leave the page (as expected), but then when someone hits back the page is brought back from the cache, but the socket is not reconnected automatially.

I added code in commit 457af82 that if it's been more than two seconds between attempting to connect and getting an error (to prevent infinite looping on connection errors), and if so, attempt to reconnect automatically.

In any case, that commit improves error handling in general by adding a  *Retry* button if something goes wrong which will restart the app.

Regards,
-Will

PS. I adjusted the credits to include your name, as I think that is appropriate. I hope that's okay. Thanks!
